### PR TITLE
Tags redesign in edit-task modal

### PR DESCRIPTION
**Pull request by mrbeardy**
## Changes
Updates the tags list in the edit-task modal to follow the design documents.

## To-do
- [x] Truncate list of tags
- [x] Add "+n more" for when user has more tags selected than max
- [X] Align tags-select dropdown with "Tags" label
- [x] Move tags select from filters into re-usable tags popup component
- [x] Use new tags popup component for box-popup when selecting tags
- [x] Add "Clear all Tags" section
- [x] Style "None"
- [x] Grow inline tags list from left to right

## Future To-do
- Allow user to edit/add tags from new tags popup component.
- Use dynamic max-width for labels depending on number of displayed labels ([Example](https://user-images.githubusercontent.com/578902/31095823-071a353e-a7b2-11e7-9b5a-128b60558ca1.png))
  *If the user has selected only 1 tag, the width of the tag will be as small as if they had selected 3 tags. Ideally the width should shrink and grow to fit the parent contents.*

## Screenshots

<div align="center">
  <h3>No tags<h3>
  

  <hr />
  <h3>Selected max tags to show</h3>
  

  <hr />
  <h3>Selected more tags than max</h3>
  

  <hr />
  <h3>Tags list popup</h3>
  

  <hr />
  <h3>Shows full tag title on hover</h3>
  <br />
  

  <hr />
  <h3>Task edit modal with tag popup</h3>
  
</div>

[//]: image
[//]: image

----
UUID: 20f6d92e-6421-41e0-9602-5b96550eba79

### CODE REVIEW

File: website/client/components/tasks/tagsPopup.vue [hunk 1]
```diff
@@ -0,0 +1,85 @@
+<template lang="pug">
+.tags-popup
+  .tags-category.d-flex
+    .tags-header
+      strong(v-once) {{ $t('tags') }}
+    .tags-list.container
+      .row
+        template
```
Comment (paglias) [inline website/client/components/tasks/tagsPopup.vue]: what's `template` here for?

**Review by paglias — CHANGES_REQUESTED** [review]

File: website/client/components/tasks/tagsPopup.vue (hunk 1 above)
Comment (mrbeardy) [reply website/client/components/tasks/tagsPopup.vue]: I've mostly copied this structure from the tags list in the filter popup in the User component, so some things might not be used.

I'm going to have a full run-through of the code at some point and clean up anything that isn't needed.

**Comment by paglias** [comment]
so this is ready, right?

**Review by paglias — APPROVED** [review]

**Comment by mrbeardy** [comment]
@paglias I think the implementation is done, but I still need to see what it looks like in the other task modals (other than habit) and also have a look at it in other browsers. I'll go do that soonish and drop a message here once it's done.

**Comment by mrbeardy** [comment]
@paglias I've had a quick look at it in firefox and in the other task types and it all seems fine (apart from an issue with the dropdown toggle arrow in firefox, but I've fixed that).

I'm happy for this to go out, I'll continue to stress-test it while working on the edit/add feature and fix any lingering issues in the next PR.

**Comment by SabreCat** [comment]
Does this address any issue we have identified over in the POST-REDESIGN label? Want to make sure I've got PRs linked to tickets where possible.

**Comment by SabreCat** [comment]
@paglias asked me to give this a test. Looks great! I have one suggestion, which is to allow a click back within the task edit modal to dismiss the tag list (it took me some puzzling to figure out I needed to click in the "+2 more" area again), but that's a minor usability improvement that doesn't need done prior to this going live.

**Comment by SabreCat** [comment]
Given there's more work planned for a future PR, I'm merging this. Thanks for the help, @mrbeardy!

**Pull request merged**